### PR TITLE
Implement cupy.linalg.pinv

### PR DESCRIPTION
**Pull request by rezoo**
This PR implements `cupy.linalg.pinv`.

### CODE REVIEW

**Comment by rezoo** [comment]
Ah I assigned @niboshi as a reviewer by mistake. Please ignore that.

File: cupy/linalg/solve.py [hunk 1]
```diff
@@ -165,7 +167,29 @@ def inv(a):
...
+    Returns:
+        cupy.ndarray: The pseudoinverse of ``a`` with dimension ``(N, M)``.
+
+    .. seealso:: :func:`numpy.linalg.pinv`
+    '''
+    u, s, vt = decomposition.svd(a, full_matrices=False)
+    cutoff = rcond * s.max()
+    s1 = 1 / s
+    s1[s <= cutoff] = 0
+    return core.dot(vt.transpose(), s1[:, None] * u.transpose())
```
Comment (okuta) [inline cupy/linalg/solve.py]: How about `return core.dot(vt.T, s1[:, None] * u.T)` ?

Comment (rezoo) [reply cupy/linalg/solve.py]: Fixed.

**Comment by okuta** [comment]
LGTM!

**Pull request merged**